src/api/index.spec.tsx

```diff
@@ -550,4 +568,120 @@ describe(__filename, () => {
       expect(query).toEqual(expect.urlWithTheseParams({ flag: 'true' }));
     });
   });
+
+  describe('createOrUpdateComment', () => {
+    const _createOrUpdateComment = (params = {}) => {
+      return createOrUpdateComment({
+        _callApi: jest.fn(),
+        addonId: 123,
+        apiState: defaultApiState,
+        cannedResponseId: undefined,
+        comment: 'Example comment',
+        commentId: undefined,
+        fileName: null,
+        line: null,
+        versionId: 321,
+        ...params,
+      });
+    };
+
+    it('can POST a comment', async () => {
+      const _callApi = jest.fn();
+      const addonId = 999;
+      const apiState = defaultApiState;
+      const cannedResponseId = undefined;
+      const comment = 'A comment about this code';
+      const fileName = null;
+      const line = null;
+      const versionId = 123;
+
+      await _createOrUpdateComment({
+        _callApi,
+        addonId,
+        apiState,
+        cannedResponseId,
+        comment,
+        fileName,
+        line,
+        versionId,
+      });
+
+      expect(_callApi).toHaveBeenCalledWith({
+        apiState,
+        bodyData: {
+          canned_response: cannedResponseId,
+          comment,
+          filename: fileName,
+          lineno: line,
+        },
+        endpoint: `reviewers/addon/${addonId}/versions/${versionId}/draft_comments`,
+        method: HttpMethod.POST,
+      });
+    });
+
+    it('can PATCH a comment', async () => {
```

Comment:
Nit: I wonder if a better name for this test would be "can update a comment" - it's more descriptive and less implementation-y.